lib/traject/sul/normalized_id.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'arclight/exceptions'
+
+module Sul
+  ##
+  # A utility class to normalize identifiers
+  class NormalizedId
+    # The Arclight to_field directive sends kwargs (:title and :repository)
+    # to this class and we use the id and title to form the collection id
+    def initialize(id, **kwargs)
+      @id = id
+      @title = kwargs[:title]
+    end
+
+    def to_s
+      normalize
+    end
+
+    private
+
+    attr_reader :id, :title
+
+    def normalize
+      # De-duplicate edge cases where id has been generated from title due to missing EAD ID
+      # E.g. don't return Université de Toulouse_universite-de-toulouse
+      # Instead, return only universite-de-toulouse.
+      # Otherwise return something like ars0009_belva-kibler-and-donald-morgan
+      [id, title].map do |string|
+        string&.truncate_words(5, omission: '')&.parameterize
+      end.compact.uniq.join('_')
```

Comment:
Sorry @marlo-longley , I didn't mean to remove the check for a valid normalized id. I think we still want to raise an error if neither the the title or id produces an id.

Reply:
The previous version had at the end of the method, which is sensible I think to prevent adding documents with empty ids to solr

```
     raise Arclight::Exceptions::IDNotFound if normalized_id.blank?

      normalized_id
```

Reply:
Sorry for all the back n forth @corylown  -- resolved.